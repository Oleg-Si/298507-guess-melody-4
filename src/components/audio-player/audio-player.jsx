import React from 'react';
import PropTypes from 'prop-types';
import withAudio from '../../hocs/with-audio/with-audio.jsx';

const AudioPlayer = (props) => {
  const {isReady, isPlaying, children, onButtonClick} = props;

  return (
    <React.Fragment>
      <button
        className={`track__button ${isPlaying ? `track__button--pause` : `track__button--play`}`}
        type="button"
        style={!isReady ? {opacity: 0.5} : {}}
        disabled={!isReady}
        onClick={onButtonClick}
      />
      <div className="track__status">
        {children}
      </div>
    </React.Fragment>
  );
};

AudioPlayer.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
  isReady: PropTypes.bool.isRequired,
  onButtonClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export {AudioPlayer};
export default withAudio(AudioPlayer);
