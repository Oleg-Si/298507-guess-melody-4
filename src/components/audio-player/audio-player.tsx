import * as React from 'react';
import withAudio from '../../hocs/with-audio/with-audio';

interface Props {
  isPlaying: boolean;
  isReady: boolean;
  onButtonClick: () => void;
  children: {};
}

const AudioPlayer: React.FC<Props> = (props: Props) => {
  const {isReady, isPlaying, children, onButtonClick} = props;

  return (
    <React.Fragment>
      <button
        className={`track__button ${isPlaying ? `track__button--pause` : `track__button--play`}`}
        type="button"
        style={!isReady ? {opacity: 0.5} : {}}
        disabled={!isReady}
        onClick={onButtonClick}
      />
      <div className="track__status">
        {children}
      </div>
    </React.Fragment>
  );
};

export {AudioPlayer};
export default withAudio(AudioPlayer);
