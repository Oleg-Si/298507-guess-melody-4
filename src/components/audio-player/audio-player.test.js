import React from 'react';
import renderer from 'react-test-renderer';
import {AudioPlayer} from './audio-player.jsx';

it(`AudioPlayer is rendered correctly`, () => {
  const tree = renderer.create(
      <AudioPlayer
        src={`url`}
        onButtonClick={() => {}}
        isPlaying={true}
        isReady={false}
      >
        <audio />
      </AudioPlayer>, {
        createNodeMock: () => {
          return {
            addEventListener: () => {}
          };
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
