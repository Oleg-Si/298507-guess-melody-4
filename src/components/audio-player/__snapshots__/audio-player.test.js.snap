// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer is rendered correctly 1`] = `
Array [
  <button
    className="track__button track__button--pause"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "opacity": 0.5,
      }
    }
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio
      autoPlay={true}
    />
  </div>,
]
`;
